clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -863,8 +864,10 @@ void EmitAssemblyHelper::CreatePasses(legacy::PassManager &MPM,
     FPM.add(createVerifierPass());
 
   // Set up the per-module pass manager.
-  if (LangOpts.SYCLIsDevice)
+  if (LangOpts.SYCLIsDevice) {
     MPM.add(createESIMDVerifierPass());
+    MPM.add(createPropagateAspectUsagePass());
```

Comment:
Could you please also include this pass into a pipeline for the new PM?

Reply:
I have a guess that it has already been added in pipeline since it was added in `PassRegistry.def` which is processed in `registerModuleAnalyses` method which is invoked from `RunOptimizationPipeline` method of `EmitAssemblyHelper`.
I've checked it with `-mllvm -debug-pass=Structure` and this pass appears in the list. I amn't sure that `RunOptimizationPipeline` is the best part since it has no common with actual optimizations. Maybe it would be better to turn it on separately in the main method in `EmitAsseblyHelper` like it is done in legacy part.